src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -150,24 +155,45 @@ def _select_extension_data(self, unit_ids):
         new_metrics = self.data["metrics"].loc[np.array(unit_ids)]
         return dict(metrics=new_metrics)
 
-    def _run(self, verbose=False):
+    def _merge_extension_data(
+        self, units_to_merge, new_unit_ids, new_sorting_analyzer, keep_mask=None, verbose=False, **job_kwargs
+    ):
+        import pandas as pd
+
+        old_metrics = self.data["metrics"]
+
+        metrics = pd.DataFrame(index=new_sorting_analyzer.unit_ids)
+        new_metrics = self._compute_metrics(new_sorting_analyzer, new_unit_ids, verbose, **job_kwargs)
+        for unit_id in new_sorting_analyzer.unit_ids:
+            if unit_id in new_unit_ids:
+                source = new_metrics
+            else:
+                source = old_metrics
+            for k in source.loc[unit_id].keys():
+                metrics.loc[unit_id, k] = source.loc[unit_id, k]
+        new_data = dict(metrics=metrics)
```

Comment:
I have the intuition that pandas could help us to do that without looping and with one line.

Reply:
maybe :-)

Reply:
done